NewHorizons/Builder/Props/ItemBuilder.cs

```diff
@@ -0,0 +1,185 @@
+
+            Delay.FireInNUpdates(() =>
+            {
+                if (item != null && !string.IsNullOrEmpty(info.pathToInitialSocket))
+                {
+                    var socketGO = planetGO.transform.Find(info.pathToInitialSocket);
+                    if (socketGO != null)
+                    {
+                        var socket = socketGO.GetComponent<OWItemSocket>();
+                        if (socket != null)
+                        {
+                            if (socket.PlaceIntoSocket(item))
+                            {
+                                // Successfully socketed
+                            }
+                            else
+                            {
+                                NHLogger.LogError($"Could not insert item {itemName} into socket at path {socketGO}");
+                            }
+                        }
+                        else
+                        {
+                            NHLogger.LogError($"Could not find a socket to parent item {itemName} to at path {socketGO}");
+                        }
+                    }
+                    else
+                    {
+                        NHLogger.LogError($"Could not find a socket to parent item {itemName} to at path {socketGO}");
+                    }
+                }
+            }, 1);
+
+            return item;
+        }
+
+        public static NHItemSocket MakeSocket(GameObject go, GameObject planetGO, Sector sector, ItemSocketInfo info)
+        {
+            var itemType = EnumUtils.TryParse(info.itemType, true, out ItemType result) ? result : ItemType.Invalid;
+            if (itemType == ItemType.Invalid && !string.IsNullOrEmpty(info.itemType))
+            {
+                itemType = EnumUtilities.Create<ItemType>(info.itemType);
+            }
+
+            var socket = go.GetAddComponent<NHItemSocket>();
+            socket._sector = sector;
+            socket._interactable = info.interactRange > 0f;
+            socket._interactRange = info.interactRange;
+
+            if (!string.IsNullOrEmpty(info.socketPath))
+            {
+                socket._socketTransform = go.transform.Find(info.socketPath);
+            }
+            if (socket._socketTransform == null)
+            {
+                var socketGO = GeneralPropBuilder.MakeNew("Socket", planetGO, sector, info, defaultParent: go.transform);
+                socketGO.SetActive(true);
+                socket._socketTransform = socketGO.transform;
+            }
+
+            socket.ItemType = itemType;
+            socket.UseGiveTakePrompts = info.useGiveTakePrompts;
+            socket.InsertCondition = info.insertCondition;
+            socket.ClearInsertConditionOnRemoval = info.clearInsertConditionOnRemoval;
+            socket.RemovalCondition = info.removalCondition;
+            socket.ClearRemovalConditionOnInsert = info.clearRemovalConditionOnInsert;
+
+            Delay.FireInNUpdates(() =>
```

Comment:
why the 2 frame delay? lol

Reply:
To make sure it runs after the similarly delayed method in for items; I only want sockets to consider themselves empty if no items get inserted into them on scene startup.